src/runner/browser-env/vite/browser-modules/driver.ts

```diff
@@ -0,0 +1,173 @@
+import {
+    WebDriverProtocol,
+    MJsonWProtocol,
+    JsonWProtocol,
+    AppiumProtocol,
+    ChromiumProtocol,
+    SauceLabsProtocol,
+    SeleniumProtocol,
+    GeckoProtocol,
+    WebDriverBidiProtocol,
+} from "@wdio/protocols";
+import { webdriverMonad, sessionEnvironmentDetector, devtoolsEnvironmentDetector } from "@wdio/utils";
+import { getEnvironmentVars } from "webdriver";
+import { SOCKET_MAX_TIMEOUT, SOCKET_TIMED_OUT_ERROR, MAX_ARGS_LENGTH } from "./constants.js";
+import { BrowserEventNames } from "./types.js";
+
+type ProtocolCommandFn = (...args: unknown[]) => Promise<unknown>;
+type PropertiesObject = Record<string, PropertyDescriptor>;
+
+const COMMANDS_HANDLED_BY_SERVER = ["debug", "saveScreenshot", "savePDF"];
```

Comment:
Most commands are initially executed on the client. Protocol commands are called inside them, which are all mocked. These commands can't be executed on client and they are immediately sent to the server.

Reply:
I would use name `SERVER_HANDLED_COMMANDS`

Reply:
> Most commands are initially executed on the client

So you are sending requests to remote grid right from the client? Does CORS mind?